types/pizzip/index.d.ts

```diff
@@ -0,0 +1,353 @@
+         * @param data the zip file
+         * @param options the options to load the zip file
+         */
+        load(data: LoadData, options?: LoadOptions): this;
+
+        /**
+         * Add (or update) a file to the zip file.
+         * You shouldn't update the data given to this method : it is kept as it so any update will impact the stored data.
+         * Throws an exception if the data is not in a supported format.
+         * @param name the name of the file. You can specify folders in the name : the folder separator is a forward slash ("/").
+         * @param data the content of the file.
+         * @param options the options.
+         */
+        file(name: string, data: Data, options?: FileOptions): this;
+
+        /**
+         * Get a file with the specified name. You can specify folders in the name : the folder separator is a forward slash ("/").
+         * @param name the name of the file.
+         */
+        file(name: string): ZipObject | null;
+
+        /**
+         * Search a file in the current folder and subfolders with a regular expression. The regex is tested against the relative filename.
+         * @param regex the regex to use.
+         */
+        file(regex: RegExp): ZipObject[];
+
+        /**
+         * Filter nested files/folders with the specified function. The predicate must return true if the file should be included, false otherwise.
+         * @param predicate the predicate to use.
+         */
+        filter(
+            predicate: (
+                /**
+                 * the filename and its path, reliatively to the current folder.
+                 */
+                relativePath: string,
+                /**
+                 * the file being tested
+                 */
+                file: ZipObject,
+            ) => boolean,
+        ): ZipObject[];
+
+        /**
+         * Create a directory if it doesn't exist, return a new PizZip object with the new folder as root.
+         * @param name the name of the directory.
+         */
+        folder(name: string): this;
+
+        /**
+         * Search a subdirectory in the current directory with a regular expression. The regex is tested against the relative path.
+         * @param regex the regex to use.
+         */
+        folder(regex: RegExp): ZipObject[];
+
+        /**
+         * Generates the complete zip file.
+         * Throws an exception if the asked type is not available in the browser,
+         * see https://github.com/open-xml-templating/pizzip/blob/master/documentation/api_pizzip/support.md
+         * @param options the options to generate the zip file
+         */
+        generate(options: GenerateOptions): ZipObject;
+
+        /**
+         * Delete a file or folder (recursively).
+         * @param name the name of the file/folder to delete.
+         */
+        remove(name: string): this;
+    }
+}
+
+declare const PizZip: {
```

Comment:
It seems like the `pizzip` library is constructable. Therefore, I would recommend following [this template](http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html) to most correctly model the underlying library.

Reply:
done